project_euler/problem_006/sol1.rb

```diff
@@ -0,0 +1,33 @@
+#Project Euler Problem 6: #https://projecteuler.net/problem=6
+
+#Sum square difference
+
+#The sum of the squares of the first ten natural numbers #is,
+#    1^2 + 2^2 + ... + 10^2 = 385
+#The square of the sum of the first ten natural numbers #is,
+#    (1 + 2 + ... + 10)^2 = 55^2 = 3025
+#Hence the difference between the sum of the squares of #the first ten
+#natural numbers and the square of the sum is 3025 - 385 = 2640.
+#Find the difference between the sum of the squares of the first one
+#hundred natural numbers and the square of the sum.
+
+def solution?()
+	num = 10
+	x = 1
+	y = 1
+	result = 1
+	gap = 3
+	while y < num
+  		x += gap
+  		gap += 2
+  		y += 1
+  		result += x
+	end
+	r_n_pow2_plus_n_pow2 = result
+	r_sum_n_pow2 = (((num / 2) + 0.5) * num) ** 2
+
+	r_sum_n_pow2 - r_n_pow2_plus_n_pow2
+end
+
+answer = solution?()
```

Comment:
To comply with https://github.com/TheAlgorithms/Ruby/blob/master/project_euler/README.md

> It should contain positional argument(s) whose default value is the question input. Example: Please take a look at [Problem 1](https://projecteuler.net/problem=1) where the question is to Find the sum of all the multiples of 3 or 5 below 1000. In this case the main solution function will be solution(limit = 1000).

```suggestion
def solution(num=10)
	x = 1
	y = 1
	result = 1
	gap = 3
	while y < num
  		x += gap
  		gap += 2
  		y += 1
  		result += x
	end
	r_n_pow2_plus_n_pow2 = result
	r_sum_n_pow2 = (((num / 2) + 0.5) * num) ** 2

	r_sum_n_pow2 - r_n_pow2_plus_n_pow2
end

answer = solution()
```

Reply:
Ok. got it.